### PR TITLE
A11y tests for Dialog and Accordion (using: axe, testing-library)

### PR DESCRIPTION
**Pull request by katarzynatobis**
Added accessibility tests for:
- `Dialog`
- `Accordion`

Full documentation [here(internal)](https://brainly.atlassian.net/wiki/spaces/DesignSystem/pages/940015977/Components+Status+IN+PROGRESS)

### CODE REVIEW

File: src/components/accordion/Accordion.spec.jsx [hunk 1]
```diff
@@ -62,24 +64,30 @@ describe('<Accordion>', () => {
   it('expands one item at a time when "allowMultiple" is set to false', () => {
     const accordion = mount(
       <Accordion>
-        <AccordionItem title="Item 1">Accordion Item Description</AccordionItem>
-        <AccordionItem title="Item 2">Accordion Item Description</AccordionItem>
-        <AccordionItem title="Item 3">Accordion Item Description</AccordionItem>
+        <AccordionItem title="Item 1" id="item1">
```
Comment (coderitual) [inline src/components/accordion/Accordion.spec.jsx:67]: @katarzynatobis do we need this change? we don't need `id` if component is uncontrolled

File: src/components/accordion/Accordion.jsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ const Accordion = ({
     invariant(
       !(isControlled && !onChange),
       // eslint-disable-next-line max-len
-      ' You provided an `expanded` prop to a Accordion without an `onChange` handler. Users won`t be able to switch between expanded/collapsed state.'
+      'You provided an `expanded` prop to a Accordion without an `onChange` handler. Users won`t be able to switch between expanded/collapsed state.'
```
Comment (coderitual) [inline src/components/accordion/Accordion.jsx:105]: one more typo: an accordion

Comment (katarzynatobis) [reply src/components/accordion/Accordion.jsx:105]: 👍🏻

File: src/components/accordion/Accordion.spec.jsx (hunk 1 above)
Comment (katarzynatobis) [reply src/components/accordion/Accordion.spec.jsx:67]: It's unnecessary, I added these identifiers for a11y testing and overlooked this change when I moved the a11y test to a separate file.

Comment (coderitual) [reply src/components/accordion/Accordion.spec.jsx]: cool, please make sure to remove ids whenever it's unnecessary to not make impression it's needed

**Review by coderitual — APPROVED** [review]

File: src/components/dialog/Dialog.a11y.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      </Dialog>
+    );
+
+    expect(dialog.getByRole('dialog')).toEqual(document.activeElement);
+  });
+
+  it('closes on Esc key', () => {
+    const dialog = render(<Dialog open>content text</Dialog>);
+
+    fireEvent.keyDown(dialog.getByRole('dialog'), esc);
```
[the remark this answers is not in the document]
Comment (katarzynatobis) [reply src/components/dialog/Dialog.a11y.spec.jsx]: I switched to user-event. 
The significant change is in 'KeyboardEvent':
- in fireEvent you specify the target,
- in user-event target must be focused before (`document.activeElement` must be set).
User can focus something in 2 ways, by clicking or tabbing. Tabbing (the real tabbing) in not available here, click will not do the job, so focus needs to be fired directly, not in the standard (user-like) way. `fireEvent.focus` just fires onfocus() event handler and do not set `document.activeElement`, but `element.focus()` does.

**Pull request merged**